seatunnel-connectors-v2/connector-cdc/connector-cdc-mysql/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mysql/source/MySqlIncrementalSourceFactory.java

```diff
@@ -99,9 +110,47 @@ public Class<? extends SeaTunnelSource> getSourceClass() {
             TableSource<T, SplitT, StateT> restoreSource(
                     TableSourceFactoryContext context, List<CatalogTable> restoreTables) {
         return () -> {
+            ReadonlyConfig config = context.getOptions();
```

Comment:
I think we should move logic into https://github.com/apache/seatunnel/blob/857aab5e8361ef12a02fddd25fe0e85d3e56f153/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/Catalog.java#L159